src/main/java/uk/ac/sanger/sccp/stan/service/SSStudyServiceImp.java

```diff
@@ -0,0 +1,118 @@
+
+/**
+ * @author dr6
+ */
+@Service
+public class SSStudyServiceImp implements SSStudyService {
+    Logger log = LoggerFactory.getLogger(SSStudyServiceImp.class);
+
+    private final Transactor transactor;
+    private final SSStudyRepo ssStudyRepo;
+    private final DnapStudyRepo dnapStudyRepo;
+
+    @Autowired
+    public SSStudyServiceImp(Transactor transactor, SSStudyRepo ssStudyRepo, DnapStudyRepo dnapStudyRepo) {
+        this.transactor = transactor;
+        this.ssStudyRepo = ssStudyRepo;
+        this.dnapStudyRepo = dnapStudyRepo;
+    }
+
+
+    @Scheduled(cron = "${spring.mlwh.schedule:-}", zone = "GMT")
+    public void scheduledUpdate() {
+        updateStudies();
+    }
+
+    @Override
+    public void updateStudies() {
+        log.info("Updating studies from mlwh");
+        Map<Integer, SSStudy> ssStudies = ssStudyRepo.loadAllSs().stream()
+                .collect(inMap(SSStudy::getId));
+        transactor.transact("updateStudies", () ->
+        {
+            Map<Integer, DnapStudy> stanStudies = stream(dnapStudyRepo.findAll()).collect(inMap(DnapStudy::getSsId));
+            update(stanStudies, ssStudies);
+            return null;
+        });
+    }
+
+    @Override
+    public List<DnapStudy> loadEnabledStudies() {
+        return dnapStudyRepo.findAllByEnabled(true);
+    }
+
+    void update(Map<Integer, DnapStudy> stanStudies, Map<Integer, SSStudy> ssStudies) {
+        List<DnapStudy> toDisable = stanStudies.values().stream()
+                .filter(ds -> ds.isEnabled() && ssStudies.get(ds.getSsId())==null)
+                .collect(toList());
+
+        List<DnapStudy> toEnable = stanStudies.values().stream()
+                .filter(ds -> !ds.isEnabled() && ssStudies.get(ds.getSsId())!=null)
+                .collect(toList());
+
+        List<DnapStudy> toRename = stanStudies.values().stream()
+                .filter(ds -> {
+                    SSStudy ss = ssStudies.get(ds.getSsId());
+                    return (ss != null && !ss.getName().equals(ds.getName()));
+                })
+                .collect(toList());
+
+        List<SSStudy> toCreate = ssStudies.values().stream()
+                .filter(ss -> stanStudies.get(ss.getId())==null)
+                .collect(toList());
+
+        Set<DnapStudy> updated = new HashSet<>(toDisable.size() + toEnable.size() + toRename.size() + toCreate.size());
+        updated.addAll(setEnabled(toDisable, false));
+        updated.addAll(setEnabled(toEnable, true));
+        updated.addAll(rename(toRename, ssStudies));
+
+        if (!updated.isEmpty()) {
+            log.info("{} studies updated", updated.size());
+            dnapStudyRepo.saveAll(updated);
+        }
+
+        if (!toCreate.isEmpty()) {
+            dnapStudyRepo.saveAll(create(toCreate));
+            log.info("{} studies created", toCreate.size());
+        }
+
+        else if (updated.isEmpty()) {
+            log.info(("No studies updated/created."));
+        }
+    }
```

Comment:
I am not very clear about the update. Would you please explain what it does? Some documentation would be great as well.

Reply:
You're right. I've added comments to this class.